app/jobs/rstuf/check_job.rb

```diff
@@ -0,0 +1,21 @@
+class Rstuf::CheckJob < Rstuf::ApplicationJob
+  RetryException = Class.new(StandardError)
+  FailureException = Class.new(StandardError)
+  retry_on RetryException, wait: :exponentially_longer, attempts: 10
+
+  queue_with_priority PRIORITIES.fetch(:push)
+
+  def perform(task_id)
+    case status = Rstuf::Client.task_status(task_id)
```

Comment:
Following my comment from `client.rb`
```suggestion
    case state = Rstuf::Client.task_state(task_id)
```